cpp-package/src/inc/brainflow_array.h

```diff
@@ -0,0 +1,468 @@
+    {
+        return at (index0);
+    }
+
+    /// access element at index
+    T &at (int index0, int index1)
+    {
+        static_assert (Dim >= 2, "BrainFlowArray dimension bounds error");
+        return origin[index0 * get_stride (0) + index1 * get_stride (1)];
+    }
+
+    /// access element at index
+    const T &at (int index0, int index1) const
+    {
+        static_assert (Dim >= 2, "BrainFlowArray dimension bounds error");
+        return origin[index0 * get_stride (0) + index1 * get_stride (1)];
+    }
+
+    /// access element at index
+    T &operator() (int index0, int index1)
+    {
+        return at (index0, index1);
+    }
+
+    /// access element at index
+    const T &operator() (int index0, int index1) const
+    {
+        return at (index0, index1);
+    }
+
+    /// access element at index
+    T &at (int index0, int index1, int index2)
+    {
+        static_assert (Dim >= 3, "BrainFlowArray dimension bounds error");
+        return origin[index0 * get_stride (0) + index1 * get_stride (1) + index2 * get_stride (2)];
+    }
+
+    /// access element at index
+    const T &at (int index0, int index1, int index2) const
+    {
+        static_assert (Dim >= 3, "BrainFlowArray dimension bounds error");
+        return origin[index0 * get_stride (0) + index1 * get_stride (1) + index2 * get_stride (2)];
+    }
+
+    /// access element at index
+    T &operator() (int index0, int index1, int index2)
+    {
+        return at (index0, index1, index2);
+    }
+
+    /// access element at index
+    const T &operator() (int index0, int index1, int index2) const
+    {
+        return at (index0, index1, index2);
+    }
+
+    /// check if array is empty
+    bool empty () const
+    {
+        return get_raw_ptr () == nullptr || get_length () == 0;
+    }
+
+    /// use it to get pointer to row in matrix or to get pointer to matrix from 3d array
+    T *get_address (int index0)
+    {
+        return &origin[index0 * get_stride (0)];
+    }
+
+    /// use it to get pointer to row in 3d array
+    T *get_address (int index0, int index1)
+    {
+        static_assert (Dim >= 2, "Dim should be >= 2");
+        return &origin[index0 * get_stride (0) + index1 * get_stride (1)];
+    }
+
+    /// use it to get pointer to row in matrix or to get pointer to matrix from 3d array
+    const T *get_address (int index0) const
+    {
+        return &origin[index0 * get_stride (0)];
+    }
+
+    /// use it to get pointer to row in 3d array
+    const T *get_address (int index0, int index1) const
+    {
+        static_assert (Dim >= 2, "Dim should be >= 2");
+        return &origin[index0 * get_stride (0) + index1 * get_stride (1)];
+    }
+
+    /// fill already preallocated buffer
+    void fill (T *ptr, int size)
+    {
+        memcpy (origin, ptr, sizeof (T) * size);
+    }
+};
+
+template <typename T, size_t Dim>
+std::ostream &operator<< (std::ostream &os, const BrainFlowArray<T, Dim> &array)
```

Comment:
thanks for implementing this, people don't always add a way to quickly output vectors and amtrics, and it really helps